lib/artemis/adapters/curb_adapter.rb

```diff
@@ -40,8 +40,16 @@ def execute(document:, operation_name: nil, variables: {}, context: {})
           easy.version = Curl::HTTP_2_0
         end
 
+        Array(callbacks&.before_request_callbacks).each do |callback|
```

Comment:
You probably don't need to wrap with an array:

```suggestion
        callbacks&.before_request_callbacks&.each do |callback|
```